src/lib/sidenav/sidenav.md

```diff
@@ -1,44 +1,133 @@
+navigation, though it can be any content) alongside some primary content. These are the sidenav and
+drawer components.
+
+The sidenav components are designed to add side content to a fullscreen app. To set up a sidenav we
+use three components: `<mat-sidenav-container>` which acts as a structural container for our content
+and sidenav, `<mat-sidenav-content>` which represents the main content, and `<mat-sidenav>` which
+represents the added side content.
 
 <!-- example(sidenav-overview) -->
 
-The sidenav and its associated content live inside of an `<mat-sidenav-container>`:
+There are also drawer components, `<mat-drawer-container>`, `<mat-drawer-content>`, and
+`<mat-drawer>`,  which are analogous to their sidenav equivalents. Rather than adding side content
+to the app as a whole, these are designed to add side content to a small section of your app. They
+support almost all of the same features, but do not support fixed positioning.
+
+<!-- example(sidenav-drawer-overview) -->
+
+### Specifying the main and side content
+
+Both the main and side content should be placed inside of the `<mat-sidenav-container>`, content
+that you don't want to be affected by the sidenav, such as a header or footer, can be placed outside
+of the container.
+
+The side content should be wrapped in a `<mat-sidenav>` element. The `position` property can be used
+to specify which end of the main content to place the side content on. `position` can be either
+`start` or `end` which places the side content on the left or right respectively in left-to-right
+languages. If the `position` is not set, the `start` end will be assumed. A
+`<mat-sidenav-container>` can have up to two `<mat-sidenav>` elements total, but only one for any
+given side.
+
+The main content should be wrapped in a `<mat-sidenav-content>`. If no `<mat-sidenav-content>` is
+specified for a `<mat-sidenav-container>`, one will be created implicitly and all of the content
+inside the `<mat-sidenav-container>` other than the `<mat-sidenav>` elements will be placed inside
+of it.
+
+<!-- example(sidenav-position) -->
+
+The following are examples of valid sidenav layouts:
+
 ```html
 <mat-sidenav-container>
-  <mat-sidenav>
-    <!-- sidenav content -->
-  </mat-sidenav>
+  <mat-sidenav>Start</mat-sidenav>
+  <mat-sidenav-content>Main</mat-sidenav-content>
+</mat-sidenav-container>
+```
 
-  <!-- primary content -->
+```html
+<mat-sidenav-container>
+  <mat-sidenav>Start</mat-sidenav>
+  <mat-sidenav position="end">End</mat-sidenav>
+  <section>Main</section>
 </mat-sidenav-container>
 ```
 
-A sidenav container may contain one or two `<mat-sidenav>` elements. When there are two 
-`<mat-sidenav>` elements, each must be placed on a different side of the container.
-See the section on positioning below.
+```html
+<mat-sidenav-container></mat-sidenav-container>
+```
+
+And these are examples of invalid sidenav layouts:
```

Comment:
Is there an overwhelming use of people doing this wrong that made us want to show people how NOT to do it?

Reply:
They illustrate the restrictions mentioned above, but for people like me who are too lazy to read words that aren't code :stuck_out_tongue: